include/zephyr/bluetooth/audio/audio.h

```diff
@@ -259,11 +269,11 @@ struct bt_codec {
 	/** Codec Specific Data count */
 	size_t   data_count;
 	/** Codec Specific Data */
-	struct bt_codec_data data[CONFIG_BT_CODEC_MAX_DATA_COUNT];
+	struct bt_codec_data data[BT_CODEC_MAX_DATA_COUNT];
```

Comment:
this might result in undefined behavior ( 0 length array inside struct),  I'd just #ifdef member instead